File: android/app/src/main/java/com/pms_parkin_mobile/receiver/BootReceiver.java
package com.pms_parkin_mobile.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.util.Log;

import com.pms_parkin_mobile.service.App;
import com.pms_parkin_mobile.service.BleScanner;

public class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();
        if (!Intent.ACTION_BOOT_COMPLETED.equals(action)
                && !"android.intent.action.QUICKBOOT_POWERON".equals(action)) return;

        App.getInstance().init(context);

        if (!App.getInstance().isServiceFlag()) {
            Log.d("BootReceiver", "ServiceFlag OFF - 서비스 미시작");
            return;
        }

        Log.d("BootReceiver", "부팅 완료 - BleScanner 재시작");
        Intent serviceIntent = new Intent(context, BleScanner.class);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            context.startForegroundService(serviceIntent);
        } else {
            context.startService(serviceIntent);
        }
    }
}

File: android/app/src/main/java/com/pms_parkin_mobile/receiver/BootReceiver.kt
package com.pms_parkin_mobile.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import android.util.Log
import com.pms_parkin_mobile.service.App
import com.pms_parkin_mobile.service.BluetoothService
import com.pms_parkin_mobile.service.SensorService

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.getAction()
        if (Intent.ACTION_BOOT_COMPLETED != action && "android.intent.action.QUICKBOOT_POWERON" != action) return

        App.instance.init(context)

        if (!App.instance.isServiceFlag) {
            Log.d("BootReceiver", "ServiceFlag OFF - 서비스 미시작")
            return
        }

        Log.d("BootReceiver", "부팅 완료 - BluetoothService / SensorService 재시작")
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            context.startForegroundService(Intent(context, BluetoothService::class.java))
            context.startService(Intent(context, SensorService::class.java))
        } else {
            context.startService(Intent(context, BluetoothService::class.java))
            context.startService(Intent(context, SensorService::class.java))
        }
    }
}
